lib/rake/install.rake

```diff
@@ -45,9 +45,8 @@ namespace :install do
         RakeUtils.python_venv_install
 
         puts CDO.dashboard_db_writer
-        if ENV['CI']
-          # Prepare for dashboard unit tests to run. We can't seed UI test data
-          # yet because doing so would break unit tests.
+        if ENV['CI'] && ENV['CI_JOB'] == 'unit_tests'
+          # Prepare for dashboard unit tests to run.
           RakeUtils.rake 'db:create db:test:prepare'
         else
           RakeUtils.rake_stream_output 'dashboard:setup_db', ([:adhoc, :development].include?(rack_env) ? '--trace' : nil)
```

Comment:
With this PR, isn't there a side effect `rake dashboard:setup_db` now gets run on ui tests whereas before it was not run? how this is possible idk, but it seems perhaps true?

I'm overall confused, but I'm slowly convincing myself that this logic does more than just NOT run `rake db:create db:test:prepare` on ui_tests.sh it also DOES run an extra `rake dashboard:setup_db`.

I'm confused because how were ui tests working WITHOUT running `dashboard:setup_db`? Yet they did seem to work.

Reply:
I checked drone logs and I think this might be an actual problem, and "quietly" adding 30 minutes to UI tests.

Reply:
Poking through, that looks right, no holes to poke.

There's a few of places I've seen (like Alex identified here!) where the unit/ui tests are kind of "bolted on" and where adapting the core logic would have perhaps been the better long-term thing to do. Someday ;-)